### ISSUE TITLE
KYMATIO_BACKEND=skcuda doesn't work

### ISSUE THREAD

**Issue by eickenberg** [kymatio/kymatio#179]
because scattering3d doesn't have that backend.

Ideally we would make scattering just not import 3D if that backend is chosen. Or we could make it be not functional and raise an error when it is executed.

 It shouldn't cause a global importerror though

**Comment by eickenberg** [comment]
fyi `KYMATIO_BACKEND_2D=skcuda` works and even `KYMATIO_BACKEND=skcuda KYMATIO_BACKEND_3D=torch` works

**Comment by janden** [comment]
I would vote for (2), that is, make it non-functional when `skcuda` backend is chosen and raise an error when instantiating (or calling `forward()`).

**Comment by janden** [comment]
@louity that would probably be the best solution, assuming it doesn't take too much time. The 15% reduction is similar to what we see in 1D and can always be improved upon later.

**Comment by louity** [comment]
@eickenberg  should I make a fork from eickenberg/kymatio or kymatio/kymatio ?

**Comment by lostanlen** [comment]
@louity the latter

**Comment by eickenberg** [comment]
Cool, thanks very much @louity for adding this backend!

As far as I know the main speed-up due to skcuda/cupy backend in the 2D version comes from the kernel implementation of the fourier downsampling.

Do you have such an implementation for the 3D code and is this already in the 15% you mention?

**Comment by louity** [comment]
Yes @eickenberg  In the 3D code it's 15% essentially because you have more complex multiplications than fourier transforms : in solid harmonic scattering transform, for 1 "convolution" with psi_l you have 2 FFT and 2L+1 complex multiplications

**Comment by lostanlen** [comment]
I can't write a skcuda backend for 3d myself, but in the interest of time, what I can do is have Kymatio return a `NotImplementedError` if `KYMATIO_BACKEND=skcuda`.

**Comment by eickenberg** [comment]
that would be OK if it throws at instantiation, but not OK if it throws that at import. It would render this environment variable useless until that backend is implemented.

So +1 for `NotImplementedError` at instantiation of a 3D object with inexistant backend.

**Comment by eickenberg** [comment]
> In the 3D code it's 15% essentially because you have more complex multiplications than fourier transforms : in solid harmonic scattering transform, for 1 "convolution" with psi_l you have 2 FFT and 2L+1 complex multiplications, so speeding up complex multiplication really speeds up the whoel thing

That makes sense, thanks

**Comment by lostanlen** [comment]
OK, I'm tagging this issue (skcuda backend by @louity) for 0.2.0 and will raise another one re: NotImplementedError
